### DISCUSSION TITLE
v3 modal has a heavy dependency on ethereum based chains

### DISCUSSION THREAD

**Discussion by darrenkel** [WalletConnect/walletconnect-monorepo#3767] [Developer Support] [answered]
Is Wallet Connect really chain agnostic? The v3 modal has a heavy dependency on wagmi, which is titled `React Hooks for Ethereum`.

**Comment by glitch-txs** [comment, marked as answer]
WalletConnect protocol is chain agnostic, Web3Modal SDK is EVM only for now, we'll add more connector in the future for other blockchains.
You can use Universal Provider + WalletConnectModal for other non-EVM blockchains
